internal/gensupport/send.go

```diff
@@ -46,8 +46,10 @@ func (e wrappedCallErr) Is(target error) bool {
 func SendRequest(ctx context.Context, client *http.Client, req *http.Request) (*http.Response, error) {
 	// Add headers set in context metadata.
 	headers := callctx.HeadersFromContext(ctx)
-	for k, v := range headers {
-		req.Header[k] = append(req.Header[k], v...)
+	for k, vals := range headers {
+		for _, v := range vals {
+			req.Header.Add(k, v)
+		}
```

Comment:
I see what you mean by the nested loop...alternatively, you could do this:
```
for k, vals := range headers {
  req.Header.Set(k, append(r.Header.Get(k), vals...))
}
```

Reply:
Thanks @noahdietz , I think I'll just stick with what I have since it's slightly clearer to look at to me if that's okay.